### DISCUSSION TITLE
外置播放器性能帧数统计表

### DISCUSSION THREAD

**Discussion by Steve-xmh** [amll-dev/applemusic-like-lyrics#206] [Show and tell]
大家可以将自己通过外置播放器获得的效果和帧数贴在这里，不论帧数高低都可以写，供其他用户参考衡量性能
（测试时请尽量减少后台占用）
建议包含以下信息：
- AMLL 版本号
- 帧数统计（设置中打开）
- 你的 CPU 和 GPU 型号
- 屏幕/窗口大小
- 一个播放器页面截图
- 是否是逐词歌词的表现
- 其他设置（背景是否静态，弹簧动画等）

目前已知会影响插件演出效果的BNCM插件如下，除非插件有所改善，否则尽量不要配合以下插件使用：

- UnblockNeteaseMusic （解灰插件）: 会导致播放进度间歇性归零，使得歌词行产生闪烁，影响动画效果（见 https://github.com/Steve-xmh/applemusic-like-lyrics/issues/210 ）
- LibFrontendPlay （前端播放库）: 疑似导致播放掉帧（见 https://github.com/Steve-xmh/applemusic-like-lyrics/discussions/206#discussioncomment-7551251 ）
- LyricBar （任务栏歌词插件）：疑似导致播放掉帧（见 https://github.com/Steve-xmh/applemusic-like-lyrics/discussions/206#discussioncomment-7775314 ）

**Comment by yrhrdcxy** [comment]
AMLL 版本号：3.00
帧数统计：188
CPU：i7-12700h
GPU：RTX 3070ti laptop
屏幕大小：1920x1080
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/62739558/44a7bb23-0e08-4ede-8c16-00f8dea22966)
是逐词歌词的表现
开启流体背景与弹簧动画

**Comment by Y-CIAO** [comment]
AMLL : 3.0.0
帧数统计 : 60(屏幕最高只有60fps)
CPU：AMD R5 4600 (移动平台)
GPU：GTX 1650 (移动平台)
屏幕/窗口大小 ：1920 x 1080
开启逐字歌词
特效全开（伪流体背景，弹簧动画，模糊，缩放）
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/132769718/df813503-4498-4e73-b902-57bf59f12feb)

**Comment by MianSoft** [comment]
AMLL 版本:3.0-dev-22c88d41
AMLLP 版本:3.0-beta-366c888c
帧数情况:163上下(切歌和辉光效果出现时会掉一下)
CPU:11th Gen Intel(R) Core(TM) i5-11400F
GPU:AMD Radeon RX 6500 XT
屏幕/窗口大小:1920x1080@165fps
开启逐词歌词(歌曲ID:1911300584)
全特效(伪流体，弹簧，模糊，缩放)
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/56493273/c41faac7-6a42-4f6c-bfab-4a94e1df2c3e)

**Reply by MianSoft** [reply]
补一个占用情况
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/56493273/7746da86-da90-49bd-8476-300ad0dc87bf)

**Comment by Xionghaizi001** [comment]
AMLL 版本:3.0-dev-22c88d41
AMLLP 版本:3.0-beta-366c888c
帧数情况: 22-113
CPU: 12th Gen Intel(R) Core(TM) i5-12450H   2.00 GHz
GPU:NVIDIA GeForce RTX 3050 Laptop GPU
屏幕/窗口大小:1920x1080@144Hz
开启逐词歌词(歌曲ID:1311347592)
全特效(伪流体，弹簧，模糊，缩放)
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/50747104/556e40ca-1bf8-4dd7-96d8-43aaca7c0eee)
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/50747104/544caf64-e9c2-4cb2-ac56-e9a30299cbb9)
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/50747104/ae398080-c7a5-4315-b9be-ea9aecaab3b2)
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/50747104/afeed75a-db1a-4640-8ff7-600120ef1ee9)

**Reply by Xionghaizi001** [reply]
- [x] 掉帧问题已成功排查：Lib Frontend Play插件引起

**Reply by Xionghaizi001** [reply]
再贴一份，硬件配置同上。
AMLL 版本:3.0.1（正式版）

帧数情况: 28-97
![image](https://github.com/Steve-xmh/applemusic-like-lyrics/assets/50747104/3472bf33-99c4-4374-bc62-ceed9568b3ec)

- [x] 排查到的掉帧原因：LyricBar插件（依赖RNP，但仅启用RNP本体不会引起过多掉帧）
- [x] 排查到的另外异常：逐词歌词随机性地延迟或提前250ms，但使用蓝牙设备不会观察到此现象

**Comment by Steve-xmh** [comment]
AMLL 版本: 3.0-beta-307a0295
AMLLP 版本: 未使用
帧数情况:在解除帧数限制但保留垂直同步的情况下范围在 73-212FPS 左右
CPU: 13th Gen Intel Core i9-13900K
GPU: NVIDIA GeForce RTX 3090
屏幕/窗口大小:3840x2160
开启逐词歌词(歌曲ID:1396524670)
全特效(伪流体，弹簧，模糊，缩放)

<img width="1920" alt="dc0d0d816e0ebe88e39e04ec16402415" src="https://github.com/Steve-xmh/applemusic-like-lyrics/assets/39523898/24a37cdd-65c9-43af-b6f2-270d11d980ad">
